docs/content/docs/build/container-states-events.md

```diff
@@ -0,0 +1,336 @@
+![A state diagram of the four possible Connection states](./images/ConnectionStates.png)
+
+#### Disconnected
+
+A container is **disconnected** if it is not in any of the other three Connection states. A container is, or becomes, disconnected in any of the following circumstances:
+
+- On the creating client, it has been created but not yet published.
+- A network problem breaks the web socket connection between the client and the Fluid service. When this happens, the Fluid runtime on the client automatically tries to reconnect, but eventually it will give up. After that, connection can only be reestablished by calling `container.connect`.
+- Your code calls the `container.disconnect` method.
+
+{{< callout important >}}
+
+Disconnection does not automatically block users from editing the shared data objects. 
+Changes they make are stored locally and will be sent to the Fluid service when connection is reestablished. 
+But these changes are *not* being synchronized with other clients while the current client is disconnected, and your application's user may not be aware of that. 
+So, you usually want to block editing if a disconnection continues for some time. 
+For more information, see [Managing connection and disconnection](#managing-connection-and-disconnection).
+
+{{< /callout >}}
+
+#### Establishing connection
+
+In this state, the client is attempting to connect to the Fluid service, but has not yet received an acknowledgement. 
+A container moves into the **establishing connection** state in any of the following circumstances:
+
+- On the creating client, your code calls the `container.attach` method. For more information, see [Publishing a container](./containers.md#publishing-a-container). This method publishes the container *and* connects the client to the service.
+- Your code calls the client's `getContainer` method on a client that has not previously been connected. For more information, see [Connecting to a container](./containers.md#connecting-to-a-container).
+- The Fluid client runtime tries to reconnect following a disconnection caused by a network problem.
+- Your code calls the `container.connect` method on a client that had become disconnected. 
+
+This process is normally very brief. When it completes, the container automatically transitions to the **catching up** state while changes made on other clients are relayed to the current client. 
+
+#### Catching up
+
+In this state, a partial connection has been established and the client is receiving changes made in other clients from the Fluid service. 
+
+Changes made locally are not uploaded to the service until catching up is complete, so the container is not considered fully connected.
+
+This state is normally very brief and the Fluid client then automatically moves to the **connected** state. 
+When the Fluid service resumes relaying data, the container automatically transitions back to the **catching up** state.
```

Comment:
Is this correct? I'm not super familiar with the area but I would have expected that catching up only occurs between **establishing connection** and **connected**. Once connected, the Fluid service is constantly relaying data, but I think a client just stays **connected** going forward.

Reply:
It was not correct. Fixed now.